launch/launch/actions/execute_process.py

```diff
@@ -717,6 +752,21 @@ async def __execute_process(self, context: LaunchContext) -> None:
                 pid, returncode, ' '.join(cmd)
             ))
         await context.emit_event(ProcessExited(returncode=returncode, **process_event_args))
+        if returncode != 0:
```

Comment:
@ivanpauno 
OK, I'll update it.

Reply:
@ivanpauno sorry, I don't see the point to relaunch the process if it exits successfully. Normally we want to respawn the process it's because the process dies unexpectedly. If the process exits successfully, it shouldn't be launched again (at least in my case). Or maybe we can have another flag to specify in which situation we want to respawn the process.

Reply:
> Normally we want to respawn the process it's because the process dies unexpectedly. If the process exits successfully, it shouldn't be launched again (at least in my case).

In that case, @hongbin-liao, feel free to submit a patch for consideration.

FWIW ROS 1 `roslaunch` respawn feature [did not pay any attention to return codes](https://github.com/ros/ros_comm/blob/12eaf946597739e833b2300ef28bb079f248a9e1/tools/roslaunch/src/roslaunch/pmon.py#L539).

Reply:
Maybe, there can be another optional argument to specify that.
Sounds like an improvement for another PR though